packages/alea-frontend/components/McqComponent.tsx

```diff
@@ -0,0 +1,409 @@
+      case 'Add References':
+        try {
+          const generationId = responseData?.generationId;
+          const response = await addReferences(currentProblem, generationId);
+          const updatedGptResponse = response?.updatedGptResponse;
+          const newProblems = parseJsonString(updatedGptResponse);
+          await extractQuestions(newProblems, response.generationObj);
+          if (currentProblems.length > 0 && generationId) {
+            await fetchAndSetVersions(generationId, setVersionsData);
+          }
+          setSelectedVersion(totalVersions + 1);
+        } catch (error) {
+          console.error('Error updating problems:', error);
+        } finally {
+          setLoading(false);
+        }
+        break;
+
+      case 'More Problems':
+        setLoading(false);
+        break;
+
+      case 'Edit':
+        setLoading(false);
+        break;
+
+      case 'Fix Distractor':
+        try {
+          const generationId = responseData?.generationId;
+          if (!generationId) return;
+          const response = await fixDistractor(currentProblem, generationId);
+          const updatedGptResponse = response?.updatedGptResponse;
+          if (!updatedGptResponse) return;
+          const newProblems = parseJsonString(updatedGptResponse);
+          await extractQuestions(newProblems, response.generationObj);
+          if (currentProblems.length > 0 && generationId) {
+            await fetchAndSetVersions(generationId, setVersionsData);
+          }
+          setSelectedVersion(totalVersions + 1);
+        } catch (error) {
+          console.error('Error updating problems:', error);
+        } finally {
+          setLoading(false);
+        }
+        break;
+      case 'Remove Ambiguity':
+        try {
+          const generationId = responseData?.generationId;
+          const response = await removeAmbiguity(currentProblem, generationId);
+          const updatedGptResponse = response?.updatedGptResponse;
+          const newProblems = parseJsonString(updatedGptResponse);
+          await extractQuestions(newProblems, response.generationObj);
+          if (currentProblems.length > 0 && generationId) {
+            await fetchAndSetVersions(generationId, setVersionsData);
+          }
+          setSelectedVersion(totalVersions + 1);
+        } catch (error) {
+          console.error('Error updating problems:', error);
+        } finally {
+          setLoading(false);
+        }
+        break;
+
+      default:
+        console.log('Unknown button clicked');
+        break;
+    }
+  };
+
+  const buttons = [
+    { name: 'More Problems', label: 'More Problems' },
+    { name: 'Add References', label: 'Add References' },
+    { name: 'Edit', label: 'Edit' },
+    { name: 'Fix Distractor', label: 'Fix Distractor' },
+    { name: 'Remove Ambiguity', label: 'Remove Ambiguity' },
+  ];
+
+  return (
+    <Box sx={{ display: 'flex', alignItems: 'flex-start', p: 3 }}>
+      <Box
```

Comment:
here you can do return (
  <>
    {loading ? (
      <Box
        sx={{
        // styles
          }}
      >
        <CircularProgress />
      </Box>
    ) : (
      // Rest of the component
    )}
  </>
);